module.nix

```diff
@@ -97,6 +97,47 @@ in
         default = { };
       };
 
+      preFormatFiles = lib.mkOption {
+        type = lib.types.attrsOf (lib.types.either lib.types.str lib.types.path);
+        description = ''
+          Files to copy into the image builder VM before disko is run.
+          This is useful to provide secrets like LUKS keys, or other files you need for formatting.
```

Comment:
Should we not have a disclaimer, that this leaks secrets into the nix store?
For image builder that runs outside of the nix store, we have a parameter that can be used instead.

Reply:
A warning that it could leak is probably a good idea. 

But the second variant of passing a shell snippet should be usable without leaking secrets into the store? i.e.

`preFormatFiles."/tmp/secret" = "cat /my-secret-out-side-nix store > $out"` shouldn't leak. $out is just a tempfile here, see prepareFiles above.

Reply:
"`cat /my-secret-outside-nix > $out`" doesn't work:

```
error: builder for '/nix/store/ragb6hgfl9wvdyb9g6721mfyymn3w0mg-myvm-disko-images.drv' failed with exit code 1;
       last 2 log lines:
       > Formatting 'main.qcow2', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=1073741824 lazy_refcounts=off refcount_bits=16
       > cat: /path/to/secret.txt: No such file or directory
```

I think this is because the `cat` command is being executed inside the builder sandbox, so it cannot access files outside the store, or something like that.

Reply:
Is it not possible to add the files outside of nix's build process (e.g. via a shell script run externally by `disko`)? I think leaking secrets like this is a huge footgun.